packages/javascript/tool/generate.ts

```diff
@@ -131,55 +140,65 @@ async function main() {
 
   async function writeMessageFunctionDocComment(definition: Condition) {
     await withDocComment(async () => {
-      await writeCommentLn(definition.description);
-      await writeCommentBlank();
       await writeCommentLn('Generate a log string for this code.');
       await writeCommentBlank();
       await writeCommentLn('This function will automatically include the log code.');
 
       for (let paramName of Object.keys(definition.message.parameters || {})) {
         const paramDef = definition.message.parameters![paramName];
-        await writeCommentParamLn(safeParam(paramName), paramDef);
+        await writeCommentParamLn(safeIdentifier(paramName), paramDef);
       }
     });
   }
 
   async function writeCodeFunctionDocComment(definition: Condition) {
     await withDocComment(async () => {
-      await writeCommentLn(definition.description);
-      await writeCommentBlank();
       await writeCommentLn('Get the code for this condition.');
     });
   }
 
+  async function writeGenericDocComment(definition: {description: string}) {
+    await withDocComment(async () => {
+      await writeCommentLn(definition.description);
+    });
+  }
+
   await writeLn('/* eslint-disable prettier/prettier */');
   await writeLn('/* eslint-disable @typescript-eslint/no-unused-vars */');
   await writeLn('/* eslint-disable @typescript-eslint/naming-convention */');
   await writeLn('// This code is automatically generated and should not be manually edited.');
   await writeLn('');
 
-  for (let [conditionCode, condition] of Object.entries(definitions.conditions)) {
-    const [systemName, sys] = Object.entries(definitions.systems).find(([_, value]) => {
-      return value.specifier == condition.system
-    })!;
-
-    const [className, cls] = Object.entries(definitions.classes).find(([_, value]) => {
-      return value.specifier == condition.class
-    })!;
-
-    await writeCondition(condition, systemName, className, conditionCode);
-  }
-
-
-  async function writeCondition(condition: Condition, systemName: string, className: string, conditionCode: string) {
-    await writeMessageFunctionDocComment(condition);
-    await scoped(`export function ${systemName}_${className}_${condition.name}(${makeParams(condition.message)}): string {`, '}', async () => {
-      await writeLn(`return \`${conditionCode} ${safeMessage(condition.message)}\`;`);
-    });
-
-    await writeCodeFunctionDocComment(condition);
-    await scoped(`export function ${systemName}_${className}_${condition.name}_code(): string {`, '}', async () => {
-      await writeLn(`return '${conditionCode}';`);
+  await scoped('export const Logs = {', '}', async () => {
```

Comment:
Curious if there's any special technique that can make all of these awaits go away, given everything is awaited. Like a magic wand that just makes it all async

Reply:
Nope, but if you do remove the awaits you get really fun output as all the IO interleaves. :-)